### PR TITLE
Use ReadOnlyCachePermissionMask in ReadOnlyJail

### PR DESCRIPTION
**Pull request by IljaN**
## Description
ReadOnlyJail should use ReadOnlyCachePermissionMask to allow file creation outside of users files dir.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/Files/Cache/Wrapper/ReadOnlyCachePermissionsMask.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	/**
+	 * @param \OCP\Files\Cache\ICacheEntry $entry
+	 * @return \OCP\Files\Cache\ICacheEntry
+	 */
+	protected function formatCacheEntry($entry) {
+		$storageId = $entry->getStorageId();
+
+		if (substr($storageId, 0, strlen('home::')) === 'home::' && $entry->getPath() === "") {
```
Comment (PVince81) [inline lib/private/Files/Cache/Wrapper/ReadOnlyCachePermissionsMask.php]: instanceof `IHomeStorage` would be way better

Comment (IljaN) [inline lib/private/Files/Cache/Wrapper/ReadOnlyCachePermissionsMask.php]: $entry is a ICacheEntry so I can`t check for instanceof IHomeStorage?

Comment (PVince81) [inline lib/private/Files/Cache/Wrapper/ReadOnlyCachePermissionsMask.php]: hmm indeed

**Comment by PVince81** [comment]
Code looks good otherwise.

Do we have integration tests for this ? @IljaN please work together with @SergioBertolinSG to have some.

**Comment by IljaN** [comment]
@SergioBertolinSG Is working on guest app inegration tests. @PVince81 Or do you mean integration tests specifically for this core feature?

**Comment by PVince81** [comment]
if it's covered by guest app it's fine for now

**Comment by PVince81** [comment]
:+1:

**Pull request merged**